feat(docs): enhance jetton page

a-bahdanau opened the pull request:
## Issue

Closes #943 and Closes https://github.com/tact-lang/tact-docs/issues/399

## Checklist

- [x] I have updated CHANGELOG.md
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

a-bahdanau committed:
feat(docs): enhance jetton page

a-bahdanau committed:
chore(docs): add pr info

a-bahdanau commented:
I left calculation of jetton address onchain (and only mentioned offchain) cause i think it will be more useful for users to have this information.

novusnota reviewed (commented):
An interesting, but slightly overengineered solution to bring Mermaid in. Cool! :)

But I'd personally get rid of the runtime JavaScript as much as possible, while making Mermaid work for the light mode _too_.

See this solution: https://astro-digital-garden.stereobooster.com/recipes/mermaid-diagrams-in-markdown/#rehype-mermaid

For it:
1. Revert the changes in `yarn.lock`, remove the `mermaid` dependency and and `yarn add rehype-mermaid`
2. Do `import rehypeMermaid from "rehype-mermaid";` and add the following `[rehypeMermaid, { strategy: "img-svg", dark: true }]` to the array of `rehypePlugins` in the `astro.config.mjs`
3. Add the custom CSS to the end of https://github.com/tact-lang/tact/blob/main/docs/src/starlight.custom.css

And then all the scripts in `plugins/` become redundant :)

UPD: So I did all my recommendations, and it's actually a worse path than you've suggested. Sorry, my bad here. But your approach also fails, since the runtime JS has to have the Mermaid dependency included as well. In the end, I think it's simpler to not add Mermaid diagrams (maybe, use some others?) for the time being until there's a streamlined solution for Starlight.

novusnota commented on docs/src/content/docs/cookbook/jettons.mdx:
```diff
@@ -154,8 +293,9 @@ fun calculateJettonWalletAddress(ownerAddress: Address, jettonMasterAddress: Add
 :::tip[Hey there!]
 
-  Didn't find your favorite example of jetton usage? Have cool implementations in mind? [Contributions are welcome!](https://github.com/tact-lang/tact/issues)
+Didn't find your favorite example of a jettons communication? Have cool implementations in mind? [Contributions are welcome!](https://github.com/tact-lang/tact-docs/issues)
```
```suggestion
Didn't find your favorite example of Jetton usage? Have cool implementations in mind? [Contributions are welcome!](https://github.com/tact-lang/tact/issues)
```

I'd revert it back since the `tact-docs` repo is slowly getting archived and the actual docs are here, in `tact`

novusnota commented:
@anton-trunov since this PR is from the org, I might not be able to edit it directly. Should we re-create those PRs branching off of this repo?

P.S.: Oh, wait, I'm able to commit suggestions via GitHub UI. Therefore, this problem is lifted :)

novusnota committed:
Update docs/src/content/docs/cookbook/jettons.mdx

Gusarich commented:
We reviewed this one already in tact-docs repo, right? Should I review it again, was anything changed from that point?

novusnota committed:
fix: remove mermaid until it gets a proper support in Starlight or when we would really need inlined diagrams

Perhaps, we can pre-compute the mermaid diagrams using it's CLI, then
embed those onto the page via a remark plugin. Unless, of course, there
would be a simpler solution available if/when we need lots of diagrams
in the future!

novusnota committed:
chore: editing busywork

novusnota committed:
Merge branch 'main' into docs/coobook-jetton

novusnota committed:
chore: final editing touches

novusnota commented:
@Gusarich yup, contents are mostly the same, although I cleaned up some parts and now it's ready to be merged, I believe. If need be, we can (and probably should) further enhance this and related pages in the future, after acquiring some feedback from chats and hackathon participants
